howso/engine/trainee.py

```diff
@@ -48,6 +53,19 @@
     Index
 )
 
+CaseIndices: TypeAlias = Iterable[Tuple[str, int]]
```

Comment:
Many of these will be useful throughout this package. Perhaps not now, but we should consider moving these into a `typing` package at the root of the `howso-engine` package.

Reply:
Technically speaking removing these from this module and adding them to a typing module in a future PR would be a MAJOR change and so I want to get it right this first time through